cmd/backup-manager/app/clean/clean.go

```diff
@@ -43,22 +51,45 @@ func (bo *Options) cleanBRRemoteBackupData(ctx context.Context, backup *v1alpha1
 		return err
 	}
 	defer s.Close()
-	iter := s.List(nil)
+
+	iter := util.ListPage(s, nil)
 	for {
-		obj, err := iter.Next(ctx)
+		// list one page of object
+		objs, err := iter.Next(ctx, pageSize)
 		if err == io.EOF {
 			break
 		}
 		if err != nil {
 			return err
 		}
-		klog.Infof("Prepare to delete %s for cluster %s", obj.Key, bo)
-		err = s.Delete(context.Background(), obj.Key)
-		if err != nil {
-			return err
+
+		// batch delete objects
+		var s3cli *s3.S3
+		var result *util.BatchDeleteObjectsResult
+		if ok := s.As(&s3cli); ok {
+			klog.Infof("Delete a page of object by S3 batch delete")
+			bucket := backup.Spec.StorageProvider.S3.Bucket
+			prefix := backup.Spec.StorageProvider.S3.Prefix
+			for i := range objs {
+				objs[i].Key = fmt.Sprintf("%s/%s", prefix, objs[i].Key)
```

Comment:
prefix may be "", "/", "//", etc. and these cases have been handled in the `cmd/backup-manager/app/util/remote.go`.

Reply:
We use `blob.PrefixBucket` to `List`, and key in result of `List`  don't contain prefix.
But `s3.S3` use complete key to delete.